src/core/pdf_manager.js

```diff
@@ -14,216 +14,203 @@
  */
 
 import {
-  createPromiseCapability, createValidAbsoluteUrl, MissingDataException,
-  shadow, unreachable, Util, warn
+  createValidAbsoluteUrl, MissingDataException, shadow, unreachable, warn
 } from '../shared/util';
 import { ChunkedStreamManager } from './chunked_stream';
 import { PDFDocument } from './document';
 import { Stream } from './stream';
 
-var BasePdfManager = (function BasePdfManagerClosure() {
-  function BasePdfManager() {
-    unreachable('Cannot initialize BaseManagerManager');
+class BasePdfManager {
+  constructor() {
+    if (this.constructor === BasePdfManager) {
+      throw new Error('Cannot initialize BasePdfManager.');
```

Comment:
Why did you replace `unreachable` with `throw new Error`?

Reply:
Whoops, I copy-pasted that from a similar piece of code in the `web` folder and we don't have `unreachable` in that context, but indeed I should be using `unreachable` here. Thank you for spotting this!